src/main/java/com/couchbase/lite/support/RemoteRequest.java

```diff
@@ -317,10 +300,66 @@ public void respondWithResult(final Object result, final Throwable error, final
                     "RemoteRequestCompletionBlock throw Exception",
                     e);
         }
-
     }
 
     public void setDontLog404(boolean dontLog404) {
         this.dontLog404 = dontLog404;
     }
+
+    public boolean isCompressedRequest() {
+        return compressedRequest;
+    }
+
+    public void setCompressedRequest(boolean compressedRequest) {
+        this.compressedRequest = compressedRequest;
+    }
+
+    protected void setBody(HttpUriRequest request) {
+        // set body if appropriate
+        if (body != null && request instanceof HttpEntityEnclosingRequestBase) {
+            byte[] bodyBytes = null;
+            try {
+                bodyBytes = Manager.getObjectMapper().writeValueAsBytes(body);
+            } catch (Exception e) {
+                Log.e(Log.TAG_REMOTE_REQUEST, "Error serializing body of request", e);
+            }
+            ByteArrayEntity entity = null;
+            if(isCompressedRequest() && bodyBytes.length > MIN_JSON_LENGTH_TO_COMPRESS){
+                entity = setCompressedBody(bodyBytes);
+            }
+            if(entity == null){
+                entity = setUncompressedBody(bodyBytes);
+            }
+            ((HttpEntityEnclosingRequestBase) request).setEntity(entity);
+        }
+    }
+    /**
+     * gzip
+     *
+     * in CBLRemoteRequest.m
+     * - (BOOL) compressBody
+     */
+    protected ByteArrayEntity setCompressedBody(byte[] bodyBytes){
+        if(bodyBytes.length < 100){
```

Comment:
Can this use MIN_JSON_LENGTH_TO_COMPRESS instead of hardcoded 100?

Also, should it throw an exception or print a warning here instead of returning null?   If it returns null, does that mean that content is never sent?